lib/phoenix/endpoint.ex

```diff
@@ -578,7 +578,20 @@ defmodule Phoenix.Endpoint do
       @doc """
       Generates a route to a static file in `priv/static`.
       """
-      def static_path(path), do: persistent!().static_path <> elem(static_lookup(path), 0)
+      def static_path(path) do
+        {path, fragment} = path_and_fragment(path)
+
+        persistent!().static_path <> elem(static_lookup(path), 0) <> fragment
+      end
+
+      defp path_and_fragment(path_incl_fragment) do
+        path_incl_fragment
+        |> String.split("#")
```

Comment:
An SVG can contain a number sign in it's id, for example `~p"/images/icons.svg#warning#small"`.

```suggestion
        |> String.split("#", parts: 2)
```